encoder/libav_encoder.cpp

```diff
@@ -145,14 +166,14 @@ void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const
 			info.colour_space->range == ColorSpace::Range::Full ? AVCOL_RANGE_JPEG : AVCOL_RANGE_MPEG;
 	}
 
-	// Apply general options.
-	encoderOptionsGeneral(options, codec_ctx_[Video]);
-
 	// Apply any codec specific options:
 	auto fn = optionsMap.find(options->libav_video_codec);
```

Comment:
I'm wondering if we should not set these codec specific options if we have a non-empty ``options->libav_video_codec_opts`` string?  This might avoid confusion with conflicting parameters?

Reply:
When an encoder codec is selected in libav, the library sets defaults and then the user overrides these with their custom settings. I kind of see the same logic with the libx264 "defaults" that are applied to the encoder with `encoderOptionsLibx264()` to get a good experience on the raspberry pi. Or any potential future defaults with `libx265` etc...
Maybe the help text should be amended to state that the custom option knobs modify the internal defaults for each codec.

Reply:
Sure, I can live with it as-is.